smarts/core/agent.py

```diff
@@ -58,6 +58,15 @@ def act(self, obs):
 
         raise NotImplementedError
 
+    def step(self, state, action, reward, next_state, done, info):
```

Comment:
This should also indicate the types of its parameters and the type of thing it returns.